pkg/ometrics/ometrics.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: Contains the main implementation of the ometrics
+// package.
+
+// Package ometrics implements a small wrapper around working with the
+// otel metrics package. It does not provide any wrappers around the
+// core types provided by otel, instead provides a way to instantiate
+// them instead.
+package ometrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/getoutreach/gobox/pkg/app"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+// ExporterType denotes the type of exporter to use for metrics.
+type ExporterType int
+
+// Contains the different types of exporters that can be used.
+const (
+	// ExporterTypePrometheus exports metrics in the prometheus format. It
+	// is the caller's responsibility to expose the metrics via an HTTP
+	// endpoint (usually through promhttp). Example implementation:
+	// https://github.com/open-telemetry/opentelemetry-go/blob/main/example/prometheus/main.go#L99
+	ExporterTypePrometheus ExporterType = iota
+
+	// ExporterTypeCollector exports metrics to the otel collector. This
+	// is akin to the "push" model of metrics, for those familiar with
+	// the prometheus model.
+	ExporterTypeCollector
+)
+
+// InitializeMeterProvider initializes the global meter provider to be
+// backed by the provided exporter.
+func InitializeMeterProvider(ctx context.Context, t ExporterType, opts ...Option) error {
```

Comment:
I would note that there technically could be an import requirement for this. Examples show the meter provider _just working_ when used as a package global, but I'm dubious and haven't looked at the actual `GetMeterProvider().Meter` code to determine if that's the case or not.

Reply:
My initial thought looking at their code was that meters created before the global provider was set would use whatever provider they were created with. However, after looking through the code more for setting the global provider, it seems they may have some means of updating the provider on pre-existing meters through some delegate concept. Would need to look even more deep into it, but it at least seems like order SHOULDN'T matter here. I will likely need to update my docs as well to account for that.